synfig-core/src/synfig/color/colorblendingfunctions.h

```diff
@@ -361,7 +359,7 @@ C blendfunc_OVERLAY(C &a,C &b,float amount)
 
 
 template <class C>
-C blendfunc_HARD_LIGHT(C &a,C &b,float amount)
```

Comment:
Is it faster passing by value instead of by reference (it should be by const reference, actually)?

Reply:
Yes its faster to pass by value in this case. It isnt const ref because the colors are actually changed but the changed colors are not used where this function is called, so ref doesn't serve a purpose.
I tried switch instead of vtable but it was slower for some reason.

Reply:
Oh, indeed! Great catch!